### ISSUE TITLE
TypeError: Cannot read property 'internal' of null

### ISSUE THREAD

**Issue by zanedev** [gatsbyjs/gatsby#8345]
## Description

Project using contentful plugin and gatsby v2 was working fine then all of a sudden started throwing this error.  I have googled around and see it could be related to the contentful content model having issues.  We are not sure what changed how can I dig into this further to find out exactly what field is having issues from contentful?  Seems like the error message should be a little better at pointing to what exactly is null.

Tried deleting cache, removing node modules and reinstalling with all fresh v2 plugins and newest gatsby no luck same problem exists.

```
Fetch Contentful data: 2165.227ms
success source and transform nodes — 2.421 s
error Cannot read property 'internal' of null


  TypeError: Cannot read property 'internal' of null
  
  - infer-graphql-input-fields.js:343 _.each
    [myproject]/[gatsby]/dist/schema/infer-graphql-input-fields.js:343:38
  
  - lodash.js:4911 
    [myproject]/[lodash]/lodash.js:4911:15
  
  - lodash.js:2996 baseForOwn
    [myproject]/[lodash]/lodash.js:2996:24
  
  - lodash.js:4880 
    [myproject]/[lodash]/lodash.js:4880:18
  
  - lodash.js:9344 Function.forEach
    [myproject]/[lodash]/lodash.js:9344:14
  
  - infer-graphql-input-fields.js:331 inferInputObjectStructureFromNodes
    [myproject]/[gatsby]/dist/schema/infer-graphql-input-fields.js:331:5
  
  - build-node-types.js:186 
    [myproject]/[gatsby]/dist/schema/build-node-types.js:186:36
  
  - Generator.next
  
  - util.js:16 tryCatcher
    [myproject]/[bluebird]/js/release/util.js:16:23
  
  - promise.js:512 Promise._settlePromiseFromHandler
    [myproject]/[bluebird]/js/release/promise.js:512:31
  
  - promise.js:569 Promise._settlePromise
    [myproject]/[bluebird]/js/release/promise.js:569:18
  
  - promise.js:614 Promise._settlePromise0
    [myproject]/[bluebird]/js/release/promise.js:614:10
  
  - promise.js:694 Promise._settlePromises
    [myproject]/[bluebird]/js/release/promise.js:694:18
  
  - async.js:138 _drainQueueStep
    [myproject]/[bluebird]/js/release/async.js:138:12
  

error UNHANDLED REJECTION


  TypeError: Cannot read property 'internal' of null
  
  - infer-graphql-input-fields.js:343 _.each
    [myproject]/[gatsby]/dist/schema/infer-graphql-input-fields.js:343:38
  
  - lodash.js:4911 
    [myproject]/[lodash]/lodash.js:4911:15
  
  - lodash.js:2996 baseForOwn
    [myproject]/[lodash]/lodash.js:2996:24
  
  - lodash.js:4880 
    [myproject]/[lodash]/lodash.js:4880:18
  
  - lodash.js:9344 Function.forEach
    [myproject]/[lodash]/lodash.js:9344:14
  
  - infer-graphql-input-fields.js:331 inferInputObjectStructureFromNodes
    [myproject]/[gatsby]/dist/schema/infer-graphql-input-fields.js:331:5
  
  - build-node-types.js:186 
    [myproject]/[gatsby]/dist/schema/build-node-types.js:186:36
  
  - Generator.next
  
  - util.js:16 tryCatcher
    [myproject]/[bluebird]/js/release/util.js:16:23
  
  - promise.js:512 Promise._settlePromiseFromHandler
    [myproject]/[bluebird]/js/release/promise.js:512:31
  
  - promise.js:569 Promise._settlePromise
    [myproject]/[bluebird]/js/release/promise.js:569:18
  
  - promise.js:614 Promise._settlePromise0
    [myproject]/[bluebird]/js/release/promise.js:614:10
  
  - promise.js:694 Promise._settlePromises
    [myproject]/[bluebird]/js/release/promise.js:694:18
  
  - async.js:138 _drainQueueStep
    [myproject]/[bluebird]/js/release/async.js:138:12

```
### Environment

```
  System:
    OS: macOS High Sierra 10.13.6
    CPU: x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.8.0 - ~/.nvm/versions/node/v10.8.0/bin/node
    Yarn: 1.9.4 - /usr/local/bin/yarn
    npm: 6.4.0 - ~/.nvm/versions/node/v10.8.0/bin/npm
  Browsers:
    Chrome: 69.0.3497.100
    Firefox: 61.0.2
    Safari: 11.1.2
  npmPackages:
    gatsby: next => 2.0.0-rc.28 
    gatsby-image: ^2.0.0-rc.2 => 2.0.6 
    gatsby-plugin-canonical-urls: next => 2.0.0-rc.2 
    gatsby-plugin-google-analytics: ^2.0.0-rc.2 => 2.0.6 
    gatsby-plugin-google-fonts: 0.0.4 => 0.0.4 
    gatsby-plugin-manifest: next => 2.0.2-rc.1 
    gatsby-plugin-netlify: next => 2.0.0-rc.6 
    gatsby-plugin-nprogress: next => 2.0.0-rc.1 
    gatsby-plugin-offline: next => 2.0.0-rc.9 
    gatsby-plugin-react-helmet: next => 3.0.0-rc.1 
    gatsby-plugin-sharp: ^2.0.0-rc.7 => 2.0.5 
    gatsby-plugin-sitemap: ^2.0.0-rc.2 => 2.0.1 
    gatsby-plugin-styled-components: ^3.0.0-rc.5 => 3.0.0 
    gatsby-remark-images-contentful: ^1.0.2 => 1.0.2 
    gatsby-remark-prismjs: next => 3.0.0-rc.2 
    gatsby-source-contentful: next => 2.0.1-rc.9 
    gatsby-source-filesystem: ^2.0.1-rc.6 => 2.0.1 
    gatsby-transformer-remark: ^2.1.1-rc.5 => 2.1.3 
    gatsby-transformer-sharp: ^2.1.1-rc.3 => 2.1.1 
  npmGlobalPackages:
    gatsby-cli: 1.1.58
    gatsby: 2.0.0-rc.0

```

**Comment by zanedev** [comment]
I was able to work around the problem by pulling down Gatsby source and adding a check for a symbol before running the _.each in `infer-graphql-input-fields` and `infer-graphql-type`.  Apparently the contentful node is resulting in a symbol when there is an issue.  This allowed me to keep working and the site loaded but I am still not sure what field exactly has the problem in contentful.  Going to keep digging on that and try to figure it out and will report back here.

After it completed this step I did notice a new error that may help me get closer to the actual node with the problem.  Maybe this warning should come before it try's to do much with the node structure and throws an unhandled exception or the exception should be handled better with a similar warning.

```
warning There are conflicting field types in your data. GraphQL schema will omit those fields.
ContentfulNavigationItem.page___NODE:
 - type: array<string>
   value: [ '771c2806-f626-50b9-9334-db1842a53481' ]
 - type: string
   value: '9c88964b-80b1-5e3a-9c04-6074166e85fb'
```

Check my PR below, not sure if this is worth merging or not or if there is a better way to approach this as a whole just thought I would post it anyway if it helps someone else in the meantime.

https://github.com/gatsbyjs/gatsby/pull/8346
